rasa/nlu/featurizers/dense_featurizer/lm_featurizer.py

```diff
@@ -42,57 +38,83 @@
 logger = logging.getLogger(__name__)
 
 
-class LanguageModelFeaturizer(DenseFeaturizer):
-    """Featurizer using transformer-based language models.
+class LanguageModelFeaturizerGraphComponent(DenseFeaturizer2, GraphComponent):
+    """A featurizer that uses transformer-based language models.
 
-    The transformers(https://github.com/huggingface/transformers) library
-    is used to load pre-trained language models like BERT, GPT-2, etc.
-    The component also tokenizes and featurizes dense featurizable attributes of
+    This component loads a pre-trained language model
+    from the Transformers library (https://github.com/huggingface/transformers)
+    including BERT, GPT, GPT-2, xlnet, distilbert, and roberta.
+    It also tokenizes and featurizes the featurizable dense attributes of
     each message.
     """
 
-    defaults = {
-        # name of the language model to load.
-        "model_name": "bert",
-        # Pre-Trained weights to be loaded(string)
-        "model_weights": None,
-        # an optional path to a specific directory to download
-        # and cache the pre-trained model weights.
-        "cache_dir": None,
-    }
+    def __init__(
+        self, config: Dict[Text, Any], execution_context: ExecutionContext,
+    ) -> None:
+        """Initializes the featurizer with the model in the config."""
+        super(LanguageModelFeaturizerGraphComponent, self).__init__(
+            execution_context.node_name, config
+        )
+        self._check_language_is_supported()
+        self._load_model_metadata()
+        self._load_model_instance(self._config["skip_model_load"])
+
+    @staticmethod
+    def get_default_config() -> Dict[Text, Any]:
+        """Returns LanguageModelFeaturizer's default config."""
+        return {
+            **DenseFeaturizer2.get_default_config(),
+            # name of the language model to load.
+            "model_name": "bert",
+            # Pre-Trained weights to be loaded(string)
+            "model_weights": None,
+            # an optional path to a specific directory to download
+            # and cache the pre-trained model weights.
+            "cache_dir": None,
+            "skip_model_load": False,
```

Comment:
we can drop the `skip_model_load` parameter and instead skip the model loading in the tests (cause this parameter must only be used there) the same way we skip e.g. the url validation [here](https://github.com/RasaHQ/rasa/blob/542c8d8fa6c29c8c55aa38f29ab50c13f204cfa9/tests/nlu/featurizers/test_convert_featurizer.py#L103)

Reply:
I'll try 🐒🩹